### PR TITLE
chore: animation-dci supports custom icon

### PR DESCRIPTION
**Pull request by kegechen**
-  animation-dci supports custom icon 
open or drop a dci icon on label to display dci icon

### CODE REVIEW

File: examples/animation-dci/main.cpp [hunk 1]
```diff
@@ -16,9 +16,11 @@
 #include <QMenuBar>
 #include <QPushButton>
 #include <QHBoxLayout>
-
+#include <qfiledialog.h>
```
Comment (zccrs) [inline examples/animation-dci/main.cpp:19]: `#include <QFileDialog>`

File: examples/animation-dci/main.cpp [hunk 2]
```diff
@@ -46,32 +48,42 @@ class IconWidget : public QMainWindow
...
         setCentralWidget(label = new QLabel(this));
+        label->setAcceptDrops(true);
         centralWidget()->setAttribute(Qt::WA_MouseTracking);
         centralWidget()->installEventFilter(this);
         statusBar()->addWidget(message = new QLabel());
 
         auto normal = new QPushButton("Normal");
         connect(normal, &QPushButton::clicked, this, [this] {
-            player.setMode(DDciIcon::Normal);
+            player.play(DDciIcon::Normal);
```
Comment (zccrs) [inline examples/animation-dci/main.cpp:58]: Why?

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/util/ddciicon.cpp [hunk 3]
```diff
@@ -1151,7 +1166,7 @@ void DDciIconImagePrivate::init()
         reader->setDevice(buffer);
         reader->setFormat(layer.format);
 
-        if (reader->supportsAnimation()) {
+        if (reader->supportsAnimation() || supportNormalImage) {
```
Comment (zccrs) [inline src/util/ddciicon.cpp:1169]: No

**Review by zccrs — CHANGES_REQUESTED** [review]

File: examples/animation-dci/main.cpp (hunk 2 above)
Comment (kegechen) [reply examples/animation-dci/main.cpp:58]: setmode会一直播放，play播放一次

File: src/util/ddciicon.cpp (hunk 3 above)
Comment (kegechen) [reply src/util/ddciicon.cpp]: ok

**Review by zccrs — DISMISSED** [review]

**Review by zccrs — APPROVED** [review]

**Pull request merged**